packages/rest/src/__tests__/integration/express.integration.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Application} from '@loopback/core';
+import {get} from '@loopback/openapi-v3';
+import {
+  Client,
+  createClientForHandler,
+  givenHttpServerConfig,
+  expect,
+} from '@loopback/testlab';
+import * as express from 'express';
+import {RestComponent} from '../../rest.component';
+import {
+  HttpRequestListener,
+  RestServer,
+  RestServerConfig,
+} from '../../rest.server';
+
+describe('HttpHandler mounted as an express router', () => {
+  let client: Client;
+  beforeEach(givenAServer);
+  beforeEach(givenClient);
+
+  context('with a simple HelloWorld controller', () => {
+    beforeEach(function setupHelloController() {
+      class HelloController {
+        @get('/hello')
+        public async greet(): Promise<string> {
+          return 'Hello world!';
+        }
+      }
+
+      givenControllerClass(HelloController);
+    });
+
+    it('handles simple "GET /hello" requests', () => {
+      return client
+        .get('/api/hello')
+        .expect(200)
+        .expect('content-type', 'text/plain')
+        .expect('Hello world!');
+    });
+
+    it('handles openapi.json', async () => {
+      const res = await client
+        .get('/api/openapi.json')
+        .set('Accept', 'application/json')
+        .expect(200);
+      expect(res.body.servers[0]).to.eql({url: '/api'});
+    });
+  });
+
+  let server: RestServer;
+  let handler: HttpRequestListener;
+
+  function givenControllerClass(
+    // tslint:disable-next-line:no-any
+    ctor: new (...args: any[]) => Object,
+  ) {
+    server.controller(ctor);
+  }
+
+  function givenClient() {
+    const app = express();
```

Comment:
I find this very confusing. The function name says that it's creating a client, but it is creating an Express server too. 

I am proposing to rework the suite setup as follows:

```ts
beforeEach(givenLoopBackApp);
beforeEach(givenExpressServer);
beforeEach(givenClient);
```

Reply:
Fixed.